### PR TITLE
Root Priming - refresh root server informations with DNSSEC verification

### PR DESCRIPTION
**Pull request by khibino**
Get the latest root server information and DNSKEY from root-hint, and use DNSSEC to verify that the content matches the trust anchor.

### CODE REVIEW

File: dnsext-full-resolver/DNS/Cache/Iterative.hs [hunk 1]
```diff
@@ -570,8 +571,12 @@ resolveJustDC dc n typ
     where
       mdc = maxNotSublevelDelegation
 
--- ドメインに対する NS 委任情報
-type Delegation = (Domain, NE DEntry)
+{- delegation information for domain -}
+type Delegation =
```
Comment (kazu-yamamoto) [inline dnsext-full-resolver/DNS/Cache/Iterative.hs:581]: Why don't you use `data`?

File: dnsext-full-resolver/DNS/Cache/RootTrustAnchors.hs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+{- import trust-anchor DS RData from
+   https://data.iana.org/root-anchors/root-anchors.xml -}
+rootSepDS :: RD_DS
+rootSepDS = RD_DS 20326 (PubAlg 8) (DigestAlg 2) digest
+  where
+    digest =
+      either (error . ("rootSepDS: bad configuration: " ++)) id
+      $ Opaque.fromBase16
+      $ fromString "E06D44B80B8F1D39A95C0B0D7C65D08458E880409BBC683457104237C7F8EC8D"
```
Comment (kazu-yamamoto) [inline dnsext-full-resolver/DNS/Cache/RootTrustAnchors.hs:19]: Probably `fromString` is not necessary with `OverloadedStrings`.

**Review by kazu-yamamoto — CHANGES_REQUESTED** [review]
Generally good. Please fix minor issues.

File: dnsext-full-resolver/DNS/Cache/RootTrustAnchors.hs (hunk 2 above)
Comment (khibino) [reply dnsext-full-resolver/DNS/Cache/RootTrustAnchors.hs]: https://github.com/kazu-yamamoto/dnsext/pull/78/commits/6b81bcb0be97efd998c7ceb2ef9a77c7ca0044b7

File: dnsext-full-resolver/DNS/Cache/Iterative.hs (hunk 1 above)
Comment (khibino) [reply dnsext-full-resolver/DNS/Cache/Iterative.hs]: https://github.com/kazu-yamamoto/dnsext/pull/78/commits/b07c5347a73d391db5b234eb517d46dcdf5f64d2

**Review by kazu-yamamoto — APPROVED** [review]
LGTM

**Pull request merged**